seed/seed.json

```diff
@@ -1327,14 +1327,35 @@
             }
         },
         {
-            "name": "EnableCrashReportingOnAndroid",
+            "name": "MetricsAndCrashSampling",
```

Comment:
upstream code creates a fallback trial with this exact name if it's not found in the seed.
So we **have** to set the name to this value in order to not trigger a dcheck.

lgtm.

https://github.com/brave/brave-browser/issues/26522
reference: https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/metrics/chrome_browser_sampling_trials.cc;l=136?q=chrome%2Fbrowser%2Fmetrics%2Fchrome_browser_sampling_trials.cc&ss=chromium%2Fchromium%2Fsrc